### PR TITLE
Merge two arrays according to constraints given.

### PR DESCRIPTION
**Pull request by priyankapiba**
<hr>

## Description 📜
Merge two arrays by satisfying given constraints
Given two sorted arrays X[] and Y[] of size m and n each where m >= n and X[] has exactly n vacant cells, merge elements of Y[] in their correct position in array X[], i.e., merge (X, Y) by keeping the sorted order.
 #7955 

<hr>

## Type of change 📝

- [ ] Au#dio (Should be in mp3 format Includes speech clarity, Concise ,Low distortion)
- [ ] Vi#deo (Animations, screen-recordings, presentations and regular explanatory films are all possibilties etc)
- [x] Documentation (Content Creation in the form of codes or tutorials)
- [ ] Other (If you choose other, Please mention changes below) 

<hr>

## Domain of Contribution 📊

- [ ] Android Dev #(Flutter)
- [ ] Android Dev #(Java)
- [ ] Android Dev #(Kotlin)
- [ ] Backend Dev #(Java)
- [ ] Backend Dev #(.NET)
- [ ] Backend Dev #(PHP)
- [ ] Backend Dev #(Python)
- [ ] C/#CPP
- [x] Competitive Programming
- [ ] Cyber #Security
- [ ] DS#A
- [ ] Data#base
- [ ] Datascience with #Python
- [ ] Datascience with #R
- [ ] Frontend Dev #HTML/CSS/JS
- [ ] Frontend Dev #React/Angular/Vue
- [ ] Go#lang
- [ ] Interview #Prep
- [ ] Java_#Domain
- [ ] JavaScript_#Domain
- [ ] ME#RN
- [ ] Machine #Learning
- [ ] Open #Source
- [ ] Python_#Domain
- [ ] Ru#st
- [ ] Statis#tics
- [ ] UI/#UX

<hr>

## Checklist ✅

- [x] I follow [Contributing Guidelines](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CONTRIBUTING.md) & [Code of conduct](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CODE_OF_CONDUCT.md) of this project.
- [x] I have performed a self-review of my own code or work.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] My changes generates no new warnings.
- [x] I'm GWOC'21 contributor

<hr>

## Screenshots / Gif (Optional) 📸

<hr>

### CODE REVIEW

**Comment by priyankapiba** [comment]
Hey @Manvi1203, my PR is under review, kindly work on a different issue.

File: Competitive_Programming/C++/Array/Merge-two-arrays.md [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+## Output:
+ 
+X[] = { 1, 2, 3, 5, 6, 8, 9, 10, 15 }
+
+## Approach:
+The idea is simple – move non-empty elements of X[] at the beginning of X[] and then merge X[] with Y[] starting from the end. The merge process is similar to the merge routine of the merge sort algorithm.
+
+## Code:
+``` Cpp
+#include <stdio.h>
```
Comment (ravireddy07) [inline Competitive_Programming/C++/Array/Merge-two-arrays.md:22]: Use `#include<bits/stdc++.h`

File: Competitive_Programming/C++/Array/Merge-two-arrays.md [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+ 
+    return 0;
+}
+```
+
+## Output:
+
+1 2 3 5 6 8 9 10 15
+
+### The time complexity of the above solution is O(m + n) and runs in constant space. Here, m and n are the size of the first and second array, respectively.
```
Comment (ravireddy07) [inline Competitive_Programming/C++/Array/Merge-two-arrays.md:105]: Add Time/Space Complexity's with Heading

**Review by ravireddy07 — CHANGES_REQUESTED** [review]

**Review by ravireddy07 — APPROVED** [review]

**Pull request merged**